src/node/audio_buffer_source.rs

```diff
@@ -436,6 +432,113 @@ impl AudioProcessor for AudioBufferSourceRenderer {
+        // basically when:
+        // - `src.start()` is called with `audio_context.current_time`,
+        //   i.e. start time is aligned with a render quantum block
+        // - the AudioBuffer was decoded w/ the right sample rate
+        // - no detune or playback_rate changes are made
+        // - loop boundaries have not been changed
+        if start_time == current_time && offset == 0. {
+            self.render_state.is_aligned = true;
+        }
+
+        // by default loop_end is 0., see AudioBufferSourceOptions
+        if loop_start != 0. || loop_end != 0. || sampling_ratio != 1. {
+            self.render_state.is_aligned = false;
+        }
+
+        // ---------------------------------------------------------------
+        // Fast track
+        // ---------------------------------------------------------------
+        if self.render_state.is_aligned {
+            if start_time == current_time {
+                self.render_state.started = true;
+            }
+
+            // check if buffer ends within this block
+            if self.render_state.buffer_time + block_duration > buffer_duration
+                || self.render_state.buffer_time + block_duration > duration
+                || current_time + block_duration > stop_time
+            {
+                let buffer_time = self.render_state.buffer_time;
+                let mut apply_offset = 0;
+
+                buffer
+                    .channels()
+                    .iter()
+                    .zip(output.channels_mut().iter_mut())
+                    .for_each(|(buffer_channel, output_channel)| {
+                        // we need to recompute that for each channel
+                        let buffer_channel = buffer_channel.as_slice();
+                        let mut start_index = (buffer_time * sample_rate).round() as usize;
+                        let end_index = if current_time + block_duration > stop_time
+                            || self.render_state.buffer_time + block_duration > duration
+                        {
+                            let dt = (stop_time - current_time)
+                                .min(duration - self.render_state.buffer_time);
+                            let end_buffer_time = self.render_state.buffer_time + dt;
+                            (end_buffer_time * sample_rate).round() as usize
+                        } else {
+                            buffer.length()
+                        };
+                        let mut offset = 0;
+
+                        for (index, o) in output_channel.iter_mut().enumerate() {
+                            let mut buffer_index = start_index + index - offset;
+
+                            *o = if buffer_index < end_index {
+                                buffer_channel[buffer_index]
+                            } else {
+                                if loop_ && buffer_index == end_index {
```

Comment:
Looping is not allowed in this code path, right? Do we need this snippet?

Reply:
No actually, we can loop in the fast path at the only condition that the loop boundaries are the default ones: i.e. `loop_start == 0.` and `loop_end == buffer.duration()`. In this case we are sure that we stay aligned with the output.

This is basically what is done in the "Simple source test without resampling (Mono|Stereo)" benches

Reply:
I agree it's a bit more code, but I think it worth the case. Not a specialist but I think that can be a pretty common situation in games for example

Reply:
Ah right, I missed that. I think that is beneficial, yes, let's keep this code path.
My comment about `offset` still stands, right?

Reply:
Yup, I actually tried to simplify but all my attempts introduced weird edge cases and crashed some of the tests...

So, I actually added a comment and changed `apply_offset` to:
```rs
let mut loop_point_index: Option<usize> = None;
```

Hopefully this is more readable, let me know